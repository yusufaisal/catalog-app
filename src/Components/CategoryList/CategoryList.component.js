import * as React from 'react';
import { View, ScrollView } from 'react-native';

import CategoryItem from '../CategoyItem/CategoyItem.component';

import styles from './CategoryList.component.styles';

const _renderCategoryItems = (props) => {
  return props.items.map(item => (
    <CategoryItem name={item.name} key={item.id}/>
  ))
};

const CategoryList = (props) => {

  return (
    <View>
      <ScrollView 
        horizontal 
        automaticallyAdjustContentInsets={false}
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={styles.scrollView}
      >
        <View style={styles.container}>
          {_renderCategoryItems(props)}
        </View>
      </ScrollView>
    </View>
    
  )
};

export default CategoryList;