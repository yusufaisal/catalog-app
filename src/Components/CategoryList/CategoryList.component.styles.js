
export default {
  container: {
    flex:1, 
    flexDirection: 'row',
    paddingHorizontal: 10
  },
  scrollView: {
  }
};