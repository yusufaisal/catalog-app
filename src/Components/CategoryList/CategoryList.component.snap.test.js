import React from 'react';
import renderer from 'react-test-renderer';

import CategoryList from './CategoryList.component';

jest
  .mock('../CategoyItem/CategoyItem.component', () => 'CategoryItem');

describe('CategoryList Component', () => {
  it('should render category list properly', () => {
    const items = [
      {
          "id": "11",
          "name": "T-Shirts"
      },{
          "id": "15",
          "name": "Women Clothing"
      },{
          "id": "13",
          "name": "Sport Shoes"
      }
    ]
    const tree = renderer
      .create(<CategoryList items={items}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});