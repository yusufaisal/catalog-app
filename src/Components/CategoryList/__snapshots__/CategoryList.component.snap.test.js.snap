// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component should render category list properly 1`] = `
<View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={Object {}}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "paddingHorizontal": 10,
          }
        }
      >
        <CategoryItem
          name="T-Shirts"
        />
        <CategoryItem
          name="Women Clothing"
        />
        <CategoryItem
          name="Sport Shoes"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
